jenkins/vm-reset

```diff
@@ -61,18 +61,35 @@ validate_args() {
 
 wait_till() {
     local vm_id=$1
-    local state=$2
+    local action_type=$2
+    local interval
+    local nr_steps
+
+    if [ $action_type == "power" ]; then
+        local state=$3
+        interval=5  # seconds
+        nr_steps=$((WAIT_TIMEOUT*60 / interval))
+    else
+        local task_id=$3 
+        interval=30  # seconds
+        nr_steps=8   # wait for task to finish
+    fi
 
-    local interval=5  # seconds
-    local nr_steps=$((WAIT_TIMEOUT*60 / interval))
     while true; do
         sleep $interval
         echo -n '.'
         ((--nr_steps > 0)) || die 'Timed out'
 
-        if [[ $(vm_power_state $vm_id) == $state ]]; then
-            echo  # add a newline after all those dots
-            return
+        if [ $action_type == "power" ]; then
```

Comment:
No need of `action_type` I think, you are not doing anything different. Pass the state as a parameter and check or the same.

Reply:
@mssawant modified the code as per comments. Please review it.